### PR TITLE
MM-19710 GH-9459

### PR DESCRIPTION
**Pull request by jaydeland**
I don’t yet have access to the main repo so I have completed number 1 and 2 on a fork.
It appears that number 3 is no longer an issue.
https://hub.docker.com/r/jaydeland/mattermost-push-proxy/tags

I have also migrated Docker to alpine:3.10.1 (since this is the latest version).
https://github.com/jaydeland/mattermost-push-proxy

CircleCI:
https://circleci.com/gh/jaydeland/mattermost-push-proxy

Ref https://github.com/mattermost/mattermost-server/issues/9459

### CODE REVIEW

**Comment by jaydeland** [comment]
CLA has also been signed

File: docker/Dockerfile [hunk 1]
```diff
@@ -1,14 +1,14 @@
...
 ENV APP_VERSION=5.9.0
 ENV PATH="/mattermost-push-proxy/bin:${PATH}"
 
 # Install some needed packages
-RUN apt-get update \
-    && apt-get -y --no-install-recommends install \
+RUN apk update \
+    && apk add \
       curl \
       wget \
```
Comment (J0WI) [inline docker/Dockerfile]: `curl` seems to be unused, `wget` is already provided by Alpine/Busybox

**Comment by J0WI** [comment]
The `mattermost-push-proxy` in the `entrypoint.sh` does not work for me, the path seems to be wrong.
Maybe you could convert this into a `CMD` and just use `exec "$@"` in the `entrypoint.sh`?
It would be good to have a tiny smoke test in the CI to catch such errors.

**Comment by jason-deland** [comment]
> The `mattermost-push-proxy` in the `entrypoint.sh` does not work for me, the path seems to be wrong.
> Maybe you could convert this into a `CMD` and just use `exec "$@"` in the `entrypoint.sh`?
> It would be good to have a tiny smoke test in the CI to catch such errors.

Turns out that go isn't in the base alpine image. However, when I add it I still get an error saying it can't find the binary.

**Comment by J0WI** [comment]
You should rather compile it against [musl libc](https://www.musl-libc.org/) than adding `libc6-compat` to an Alpine image.

**Comment by jason-deland** [comment]
> You should rather compile it against [musl libc](https://www.musl-libc.org/) than adding `libc6-compat` to an Alpine image.

I have updated the Makefile to include the CC_FOR_TARGET ENV Var and this worked on my local testing without libc6-compat.

@J0WI Can you please confirm?

File: docker/Dockerfile [hunk 2]
```diff
@@ -1,14 +1,12 @@
...
-RUN apt-get update \
-    && apt-get -y --no-install-recommends install \
-      curl \
-      wget \
-      netcat \
+RUN apk update \
+    && apk add \
+      netcat-openbsd \
       ca-certificates \
     && rm -rf /var/lib/apt/lists/* /var/cache/apt/archives/*.deb /var/cache/apt/archives/partial/*.deb /var/cache/apt/*.bin
```
Comment (J0WI) [inline docker/Dockerfile]: This line is Debian/Ubuntu specific

File: docker/Dockerfile [hunk 3]
```diff
@@ -1,14 +1,12 @@
...
 ENV PATH="/mattermost-push-proxy/bin:${PATH}"
 
 # Install some needed packages
-RUN apt-get update \
-    && apt-get -y --no-install-recommends install \
-      curl \
-      wget \
-      netcat \
+RUN apk update \
+    && apk add \
```
Comment (J0WI) [inline docker/Dockerfile]:
```suggestion
    && apk add --no-cache\
```

File: docker/Dockerfile [hunk 4]
```diff
@@ -1,14 +1,12 @@
...
 ENV APP_VERSION=5.9.0
 ENV PATH="/mattermost-push-proxy/bin:${PATH}"
 
 # Install some needed packages
-RUN apt-get update \
-    && apt-get -y --no-install-recommends install \
-      curl \
-      wget \
-      netcat \
+RUN apk update \
```
Comment (J0WI) [inline docker/Dockerfile]:
```suggestion
RUN apk add --no-cache\
```
(update is obsolete with `--no-cache`)

Comment (jason-deland) [reply docker/Dockerfile]: Good point!

Comment (jaydeland) [reply docker/Dockerfile]: Good point!

File: docker/Dockerfile (hunk 2 above)
Comment (jaydeland) [reply docker/Dockerfile]: Removed

Comment (jaydeland) [reply docker/Dockerfile]: Removed

**Comment by jaydeland** [comment]
@J0WI - I removed the extra Debian specific lines and tested the build locally. I think we are good at this point for further testing to validate the service is working as expected.

**Comment by cpanato** [comment]
thanks @jaydeland !

**Pull request merged**